### PR TITLE
Add Reviewer Assignment to Funding Applications

### PR DESCRIPTION
**Pull request by laHermes**
Adds the ability to assign program reviewers and milestone reviewers to funding applications directly from the application list.

**Key Changes:**
- Added `appReviewers` and `milestoneReviewers` fields to `IFundingApplication` type
- Created `ReviewerAssignmentDropdown` component for multi-select reviewer assignment
- Added `useReviewerAssignment` hook for assignment logic and error handling
- Created `application-reviewers.service` for API integration
- Updated `ApplicationList` to display reviewer columns when reviewers are available
- Enhanced `MultiSelectDropdown` with loading states and improved UX
- Added comprehensive test coverage for the service

**Features:**
- Multi-select dropdowns for assigning reviewers per application
- Separate columns for App Reviewers and Milestone Reviewers
- Real-time updates with optimistic UI feedback
- Graceful error handling with user-friendly messages
- Columns only appear when reviewers are configured for the program

**Testing:**
- Added unit tests for `application-reviewers.service` covering success cases, error handling, and edge cases

**Screenshots:**

Xnip2025-12-11_16-25-45
Xnip2025-12-11_16-35-50

## Summary by CodeRabbit

* **New Features**
  * Reviewer assignment UI: app & milestone reviewer columns with multi-select dropdowns, loading/error states, and backend assignment with feedback.
  * Consolidated table/row components for cleaner rendering, sorting, status actions, and memoized performance.

* **Data & APIs**
  * New reviewer service and assignment logic; application types extended to include reviewer fields; query keys added for funding data.

* **Accessibility & UX**
  * Improved table header semantics, ARIA attributes, and reviewer header indicators.

* **Tests**
  * Extensive unit and integration tests covering dropdowns, assignment flows, service calls, and error scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: services/application-reviewers.service.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const apiClient = createAuthenticatedApiClient(API_URL, 30000);
+
+// Add response interceptor for error handling
+apiClient.interceptors.response.use(
+  (response) => response,
+  (error) => {
+    console.error("Application Reviewers API Error:", error.response?.data || error.message);
+    throw error;
+  }
+);
```
Comment (amaurymagalhaes) [inline services/application-reviewers.service.ts]: Can we use our standard fetchData?

File: services/application-reviewers.service.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ */
+export const applicationReviewersService = {
+  /**
+   * Assign reviewers to an application
+   */
+  async assignReviewers(
+    applicationId: string,
+    request: AssignApplicationReviewersRequest
+  ): Promise<void> {
+    await apiClient.put(`/v2/funding-applications/${applicationId}/reviewers`, request);
```
Comment (amaurymagalhaes) [inline services/application-reviewers.service.ts]: please always use INDEXER constant to refer to endpoint, also use fetchData

File: services/application-reviewers.service.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  async getAssignedReviewers(applicationId: string): Promise<{
+    appReviewers: string[];
+    milestoneReviewers: string[];
+  }> {
+    try {
+      const response = await apiClient.get<{
+        appReviewers?: string[];
+        milestoneReviewers?: string[];
+      }>(`/v2/funding-applications/${applicationId}/reviewers`);
```
Comment (amaurymagalhaes) [inline services/application-reviewers.service.ts]: please always use INDEXER constant to refer to endpoint, also use fetchData

**Review by amaurymagalhaes — CHANGES_REQUESTED** [review]
Couldn't test the functionality because of the backend part is missing.
